api/applicationserver_integrations_storage.proto

```diff
@@ -50,9 +51,9 @@ message GetStoredApplicationUpRequest {
 
   // Limit number of results.
   google.protobuf.UInt32Value limit = 4;
-  // Query upstream messages after this timestamp only.
+  // Query upstream messages after this timestamp only. Cannot be used in conjunction with last.
   google.protobuf.Timestamp after = 5 [(gogoproto.nullable) = true, (gogoproto.stdtime) = true];
```

Comment:
Maybe this is a good moment to get rid of the `gogoproto` tags here and explicitly check for empty value in code.